### PR TITLE
Added in `default="1"` for `Choice(title="bleh",value="1")`

### PR DESCRIPTION
**Pull request by WilliamStam**
the way "default" is handled currently isnt very intuitive. you need to pass a Choice object to it if you want it to use that choice's value

this PR lets you supply the default as just the "value" of the choice. 

as per 

https://github.com/tmbo/questionary/issues/124 and https://github.com/tmbo/questionary/issues/57 

(sorry not sure how to add tests for it, still noob)

### CODE REVIEW

**Comment by kiancross** [comment]
I don't think this works. A failing example is below:

```python3
import questionary

choice = questionary.Choice("Bar")

question = questionary.select(
    "What do you want to do?", choices=["Foo", choice], default=choice
)

question.ask()
```

I've also added some additional tests in #131 which will catch some breaking changes, such as when the value of a `Choice` and a string are the same.

File: questionary/prompts/common.py [hunk 1]
```diff
@@ -217,7 +217,14 @@ def __init__(
         self.default = default
         self.pointer = pointer
 
-        if default is not None and default not in choices:
+        choices_values = list()
```
Comment (kiancross) [inline questionary/prompts/common.py:220]:
```suggestion
        choices_values = []
```

**Comment by WilliamStam** [comment]
i see why its causing that issue and ive now gotten a "fix" for it. but i now also see that it causes another issue with "initial_choice"

no errors, just the option isn't selected. 
````
questionary.checkbox(
        "What do you want to do?",
        choices=[
            questionary.Choice(
                title="First option",
                value="1"
            ),
            questionary.Choice(
                title="Second option",
                value="2"
            ),
            questionary.Choice(
                "third option"
            ),
            forth_option,
            'Make a reservation',
            'Ask for opening hours'
        ]
 
    ).ask()
````
default="2", # works
default='third option', #works
default="Make a reservation", # works
default=forth_option, # doesnt work

initial_choice='2', # works
initial_choice='third option', # works 
initial_choice='Make a reservation', # works 
initial_choice=forth_option, # works 

so its just to get default=forth_option, to work now :( 

cleaning up code. i dont know how to "merge" your tests into my fork so for now i just copied them over and replaced mine localy and ran them (bitches about code style but the tests work fine.

**Comment by WilliamStam** [comment]
image

it does actualy seem that `default=forth_option,` doesnt actualy work as intended on the main repo common.py either (it does nothing)

image

test with this

```
forth_option = questionary.Choice(
        "forth option"
    )

    questionary.checkbox(
        "What do you want to do?",
        choices=[
            questionary.Choice(
                title="First option",
                value="1"
            ),
            questionary.Choice(
                title="Second option",
                value="2"
            ),
            questionary.Choice(
                "third option"
            ),
            forth_option,
            'Make a reservation',
            'Ask for opening hours'
        ]
        ,
        # default=forth_option,
        # initial_choice=forth_option,
        # default="2",
        default=forth_option,
        # default="Make a reservation",
    ).ask()
```

**Comment by WilliamStam** [comment]
is there ever a case where default would for instance be a list? as in default = [choice1,choice2] ? im blinly walking into this assuming that that isnt a valid use case for this. if you want to have multiple items checked then 
```
"choices":[
    Choice(.... default="t")
]
```
right?

**Comment by WilliamStam** [comment]
at least with my PR we wont have to do stuff like this any more

```
 enabled_choices = [
            questionary.Choice(
                title="Enable",
                value="1",
                checked=True if default == '1' else False,
            ),
            questionary.Choice(
                title="Disable",
                value="0",
                checked=True if default != '1' else False
            ),
        ]
        enabled_default = None
        for row in enabled_choices:
            if row.checked:
                enabled_default = row

        selected = questionary.unsafe_prompt([
            {
                "type": "select",
                "name": "enabled",
                "message": "Enable/Disable the controller",
                "choices": enabled_choices,
                "default":enabled_default
            },
```

**Comment by WilliamStam** [comment]
i fixed the codestyle thing. not sure my environment is setup correctly since `pytest --pycodestyle --cov questionary -v` doesnt pick it up locally

**Comment by kiancross** [comment]
It looks like there is a bug somewhere in `master` - having a look now.

> i fixed the codestyle thing. not sure my environment is setup correctly since `pytest --pycodestyle --cov questionary -v` doesnt pick it up locally

You can run `make formatter` to make sure all the code is correctly formatted.

> i dont know how to "merge" your tests into my fork so for now i just copied them over and replaced mine localy and ran them

I've merged my tests with this PR now.

**Comment by WilliamStam** [comment]
is there something i can do to help with this PR? (i'll try have a look to see if i can try find the "forth option" bug soonish. (having a weekend for a change....)

**Comment by kiancross** [comment]
I haven't had time to have a look at the PR properly again yet - will have time sometime in the next week.

For testing, you can see [here](https://coveralls.io/builds/40721482/source?filename=questionary%2Fprompts%2Fcommon.py#L240) that some of the code you have written is not being covered by unit tests.

**Comment by WilliamStam** [comment]
ive added a test in that i "think" might solve that. if anyone is able to help me with this it will be appreciated

**Comment by WilliamStam** [comment]
well that just screwed it up. i give up. 

heres the "changes"

in the select test
```
def test_select_initial_choice_with_value():
    message = "Foo message"
    choice = Choice(title="bazz",value="bar")
    kwargs = {"choices": ["foo", choice], "default": "bar"}
    text = KeyInputs.ENTER + "\r"

    result, cli = feed_cli_with_input("select", message, text, **kwargs)
    assert result == "bar"
```

and for the PR part before merging and stuff ended up being dirty (i honestly just give up on submitting pr's to any projects :( )

prompts/common.py 220ish 
```
        if isinstance(default, Choice):
            default = default.value

        choices_values = []
        for choice in choices:
            if isinstance(choice, Choice):
                choices_values.append(choice.value)

        if default is not None and default not in choices and default not in choices_values:
            raise ValueError(
                f"Invalid `default` value passed. The value (`{default}`) "
                f"does not exist in the set of choices. Please make sure the "
                f"default value is one of the available choices."
            )

        if initial_choice is None:
            pointed_at = None
        elif initial_choice in choices:
            pointed_at = choices.index(initial_choice)
        elif initial_choice in choices_values:
            for k, choice in enumerate(choices):
                if isinstance(choice, Choice):
                    if choice.value == initial_choice:
                        pointed_at = k
                        break
```

**Comment by kiancross** [comment]
Sorry for the delay in looking at this. The two changes are in quite core places in the code, so I want to ensure I take a close look at them before I merge in.

When I go through this I can fix any of the issues with the failing tests :) Hopefully in the next couple of weeks.

**Comment by WilliamStam** [comment]
i'll try see if i can start over with this PR and not screw it up so much. not totally familiar with the testing stuff in python tho. so was sort of looking for help with that. (new to py). something was definitely screwy with my black install or something cause it wanted to format a lot more lines than it needed to.  (im on windows so "make formatter" doesnt work yet, but guessing i need to WSL it if im ever gonna get to join the cool kids)

ive tried a bunch of things with the above code and they "seem" to work ok but thats before the fixes for "option 4" thing:

```
forth_option = questionary.Choice(
     "forth option"
)

default=forth_option,
```

https://github.com/tmbo/questionary/issues/132

**Comment by WilliamStam** [comment]
thank you :)

**Review by tmbo — APPROVED** [review]
looks good - thanks a lot for this improvement :+1:

**Pull request merged**